hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerCall.java

```diff
@@ -166,7 +166,7 @@ private void release(int mask) {
   }
 
   @Override
-  public void cleanup() {
+  public synchronized void cleanup() {
```

Comment:
Does this really requires to be synchronized? If so, you might also want to synchronize releaseByWAL method?

Reply:
This method can be called by Handler or Netty Worker concurrently.   
But releaseByWAL do not.

Reply:
But I think the implementation is thread safe?

Reply:
OK, i make a mistake, it is thread safe now.